collectors/ebpf.plugin/ebpf.c

```diff
@@ -1544,6 +1567,8 @@ int main(int argc, char **argv)
             NULL, NULL, ebpf_modules[EBPF_MODULE_DISK_IDX].start_routine},
         {"EBPF MOUNT" , NULL, NULL, 1,
             NULL, NULL, ebpf_modules[EBPF_MODULE_MOUNT_IDX].start_routine},
+        {"EBPF HARDIRQ" , NULL, NULL, 1,
```

Comment:
Hello,

Are you sure we will have one for `hardirq` and another for `softirq`? Can we have both in one unique thread?

Reply:
@thiagoftsm It all comes down to how we look at our architecture. Theoretically we could combine every thread we currently have into one thread and find a way to make it work. So the question is, how is the border decided?

If we base it off of unique kernel collectors in the kernel-collections repo, then it should be 1 thread for each, which is the model I've followed here.

Reply:
My question was based in the fact that both are working with IRQ, take a look at `filesystem` thread, for example, there we are monitoring `ext4`, `nfs`, `xfs`, `zfs` and `btrfs`. 

I did not take a look inside both eBPF programs that you are working yet, my doubt has relationship with the `IRQ` that both has in its name. If they are not bringing the same charts, so we must keep them separated.

Reply:
Ah I see, okay!

The main difference between the two, is that hard IRQ IDs are not set in stone, but the software IRQs are. The actual charts for each will be separate charts, but the parameters and type of data of the chart will be the same. Would that warrant making a single thread do the collection then? (my own opinion is that there is no issue here, so I'm happy to structure this so it combines them).

Reply:
Considering your description, to unify both is the simplest way to bring both and save your time. :)

Reply:
Alright sounds good!

Reply:
Ok after looking at this again, I honestly don't think we should combine them, as they are only coincidentally similar in name, but for example their tracepoints are different, the labels are static for one and dynamic for the other, and there will likely be more differences I encounter as I develop them.